### ISSUE TITLE
32-bit build fails

### ISSUE THREAD

**Issue by kankaristo** [g-truc/glm#528]
32-bit builds of GLM seem to fail:
```cpp
In file included from ../thirdparty/glm/detail/../detail/func_common.hpp:426:0,
                 from ../thirdparty/glm/detail/../detail/func_geometric.inl:5,
                 from ../thirdparty/glm/detail/../detail/func_geometric.hpp:113,
                 from ../thirdparty/glm/detail/../geometric.hpp:6,
                 from ../thirdparty/glm/detail/func_matrix.inl:4,
                 from ../thirdparty/glm/detail/func_matrix.hpp:149,
                 from ../thirdparty/glm/detail/type_mat2x2.inl:4,
                 from ../thirdparty/glm/detail/type_mat2x2.hpp:182,
                 from ../thirdparty/glm/mat2x2.hpp:6,
                 from ../thirdparty/glm/glm.hpp:71:
../thirdparty/glm/detail/../detail/func_common.inl:193:44: error: wrong number of template arguments (4, should be 5)
  struct compute_sign<T, P, vecType, Aligned>
                                            ^
../thirdparty/glm/detail/../detail/func_common.inl:183:9: note: provided for 'template<class T, glm::precision P, template<class, glm::precision <anonymous> > class vecType, bool isFloat, bool Aligned> struct glm::detail::compute_sign'
  struct compute_sign
         ^~~~~~~~~~~~
In file included from ../thirdparty/glm/gtc/../detail/../detail/func_common.hpp:426:0,
                 from ../thirdparty/glm/gtc/../detail/../detail/func_geometric.inl:5,
                 from ../thirdparty/glm/gtc/../detail/../detail/func_geometric.hpp:113,
                 from ../thirdparty/glm/gtc/../detail/../geometric.hpp:6,
                 from ../thirdparty/glm/gtc/../detail/func_matrix.inl:4,
                 from ../thirdparty/glm/gtc/../detail/func_matrix.hpp:149,
                 from ../thirdparty/glm/gtc/../detail/type_mat3x3.inl:4,
                 from ../thirdparty/glm/gtc/../detail/type_mat3x3.hpp:189,
                 from ../thirdparty/glm/gtc/../mat3x3.hpp:6,
                 from ../thirdparty/glm/gtc/quaternion.hpp:18:
../thirdparty/glm/gtc/../detail/../detail/func_common.inl:193:44: error: wrong number of template arguments (4, should be 5)
  struct compute_sign<T, P, vecType, Aligned>
                                            ^
../thirdparty/glm/gtc/../detail/../detail/func_common.inl:183:9: note: provided for 'template<class T, glm::precision P, template<class, glm::precision <anonymous> > class vecType, bool isFloat, bool Aligned> struct glm::detail::compute_sign'
  struct compute_sign
         ^~~~~~~~~~~~
```

The build is failing with GCC 5.2.1 on Linux and with MinGW 6.1.0 on Windows/MSYS2.

Removing the code inside the `#if GLM_ARCH == GLM_ARCH_X86` section seems to fix the build.

**Comment by kankaristo** [comment]
Create a PR at https://github.com/g-truc/glm/pull/529, if removing the code section is okay (it may break something, I'm not sure).

**Comment by Groovounet** [comment]
This issue is fixed in GLM 0.9.8 and master branches.

Thanks for reporting,
Christophe
